modules/mobile/spec/request/letters_lighthouse_request_spec.rb

```diff
@@ -0,0 +1,117 @@
+# frozen_string_literal: true
+
+require 'rails_helper'
+require_relative '../support/iam_session_helper'
+require_relative '../support/matchers/json_schema_matcher'
+
+RSpec.describe 'letters', type: :request do
+  include JsonSchemaMatchers
+
+  let(:rsa_key) { OpenSSL::PKey::RSA.generate(2048) }
+  let(:letters_body) do
+    {
+      'data' => {
+        'id' => '3097e489-ad75-5746-ab1a-e0aabc1b426a',
+        'type' => 'letters',
+        'attributes' => {
+          'letters' =>
+            [
+              {
+                'name' => 'Commissary Letter',
+                'letterType' => 'commissary'
+              },
+              {
+                'name' => 'Proof of Service Letter',
+                'letterType' => 'proof_of_service'
+              },
+              {
+                'name' => 'Proof of Creditable Prescription Drug Coverage Letter',
+                'letterType' => 'medicare_partd'
+              },
+              {
+                'name' => 'Proof of Minimum Essential Coverage Letter',
+                'letterType' => 'minimum_essential_coverage'
+              },
+              {
+                'name' => 'Service Verification Letter',
+                'letterType' => 'service_verification'
+              },
+              {
+                'name' => 'Civil Service Preference Letter',
+                'letterType' => 'civil_service'
+              },
+              {
+                'name' => 'Benefit Summary and Service Verification Letter',
+                'letterType' => 'benefit_summary'
+              },
+              {
+                'name' => 'Benefit Verification Letter',
+                'letterType' => 'benefit_verification'
+              }
+            ]
+        }
+      }
+    }
+  end
+
+  before do
+    allow(File).to receive(:read).and_return(rsa_key.to_s)
+    allow_any_instance_of(IAMUser).to receive(:icn).and_return('24811694708759028')
+    user = build(:iam_user)
+    iam_sign_in(user)
+    Flipper.enable(:mobile_lighthouse_letters, user)
+  end
+
+  before(:all) do
+    @original_cassette_dir = VCR.configure(&:cassette_library_dir)
+    VCR.configure { |c| c.cassette_library_dir = 'modules/mobile/spec/support/vcr_cassettes' }
+  end
+
+  after(:all) { VCR.configure { |c| c.cassette_library_dir = @original_cassette_dir } }
+
+  describe 'GET /mobile/v0/letters' do
+    context 'with a valid lighthouse response' do
+      it 'matches the letters schema' do
+        VCR.use_cassette('lighthouse_letters/letters_200', match_requests_on: %i[method uri]) do
+          get '/mobile/v0/letters', headers: iam_headers
+          expect(response).to have_http_status(:ok)
+          expect(JSON.parse(response.body)).to eq(letters_body)
+          allow(File).to receive(:read).and_call_original
```

Comment:
I'm a little confused by this. Is this intended to overwrite the stub in the before block?

Reply:
I was banging by head against this one for a while. the schema validation code, calls a `read` on a `File` and kept failing on me. It was only when I realized that it was returning the rsa key from above instead of the actual schema json file that I realized that the stub needed to be reversed. I tried playing with more elegant ways of doing this (like using `.with`) but couldn't find another solution.

Reply:
That makes sense. Given the way this spec file is structured, couldn't you just move the stub from the before block above into the error handling context. Isn't that where the stub is actually needed?

Reply:
sorry, forgot to respond to this one. you're right. it shouldn't be nested so far down but when I try to bring it up, it starts failing again and this time I'm not sure exactly why. there must be other File reads being done with VCR. I'm gonna keep it there for now but look more into it for the future letters PR that I'll be creating today or tomorrow. I'm just surprised we haven't ran into this issue before since we often compare the schemas and use that RSA file read stub